src/brsTypes/components/RoSGNode.ts

```diff
@@ -435,30 +436,53 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         return this.getMethod(index.value) || BrsInvalid.Instance;
     }
 
-    set(index: BrsType, value: BrsType, alwaysNotify: boolean = false, kind?: FieldKind) {
+    set(index: BrsType, value: BrsType) {
         if (index.kind !== ValueKind.String) {
             throw new Error("RoSGNode indexes must be strings");
         }
 
         let mapKey = index.value.toLowerCase();
-        let fieldType = kind || FieldKind.fromBrsType(value);
         let field = this.fields.get(mapKey);
 
         if (!field) {
-            // RBI does not create a new field if the value isn't valid.
-            if (fieldType) {
-                field = new Field(value, fieldType, alwaysNotify);
-                this.fields.set(mapKey, field);
-            }
+            // TODO: change to using interpreter.stderr
```

Comment:
If having this creates issues int the current PR (which I think it's breaking unit tests right now) I'm ok with opening a github issue and tackling this work in a separate PR

Reply:
yep, did it